legend-shared-utility-interfaces/src/main/java/org/finos/legend/shared/test/junit/DatabaseIntegrationTest.java

```diff
@@ -0,0 +1,24 @@
+// Copyright 2023 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.shared.test.junit;
+
+/*
+    Marker interface to be used with Junit Categories. See https://junit.org/junit4/javadoc/4.12/org/junit/experimental/categories/Categories.html
+
+    Used to indicate integration tests that test database functionality e.g. SQL generation, authentication etc
+ */
+public interface DatabaseIntegrationTest
```

Comment:
What is the difference between this and IntegrationTest?  Why we need both?

Reply:
One is more specific than the other ? For e.g for SQL tests, we want the surefire plugin to select a set of tests.  I'm thinking of even more fine grained categories.

Reply:
Maybe we annotate the Junit categoies as @ExperimentalApi :-D

Reply:
What is the value for that?  Like, where you think we will run Integration test cases, but not SQL or vice-versa?

We will have an increase likelihood of not running test cases if we make this a complicated set of toggles.

We should be looking at this from the "running any test" POV so that we can try to make as intelligent decisions as possible when scheduling.  The closest you get to what is happening on the test, the harder will get to make further refactoring on how things are executed.